ElectronicObserver/Window/Tools/ExpeditionRecordViewer/ExpeditionRecordViewModel.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Linq;
+using ElectronicObserver.Database.Expedition;
+using ElectronicObserver.KancolleApi.Types.ApiReqMission.Result;
+using ElectronicObserver.Window.Tools.SortieRecordViewer;
+using ElectronicObserverTypes;
+using System.Text.Json;
+using System.Collections.Generic;
+using ElectronicObserver.Data;
+
+namespace ElectronicObserver.Window.Tools.ExpeditionRecordViewer;
+
+public class ExpeditionRecordViewModel
+{
+	public ExpeditionRecord? Model { get; }
+	public int? Id => Model?.Id;
+	public DateTime ExpeditionStart { get; }
+	public IFleetData? Fleet { get; }
+	public List<int>? ItemList { get; }
+	public object? MaterialList { get; }
+	public int MaterialFuel { get; }
+	public int MaterialAmmo { get; }
+	public int MaterialSteel { get; }
+	public int MaterialBaux { get; }
+	public int? ItemOneID { get; }
+	public string? ItemOneName { get; }
+	public int? ItemOneCount { get; }
+	public int? ItemTwoID { get; }
+	public int? MapAreaID { get; }
+	public string? DisplayID { get; }
+	public string? ItemTwoName { get; }
+	public int? ItemTwoCount { get; }
+	public string? ClearResult { get; }
+	public string? ItemOneString { get; }
+	public string? ItemTwoString { get; }
+
+	public ExpeditionRecordViewModel(ExpeditionRecord record, ApiReqMissionResultResponse response, DateTime expeditionStart)
+	{
+		if (response is null) return;
+		Model = record;
+		ExpeditionStart = expeditionStart.ToLocalTime();
+		MapAreaID = KCDatabase.Instance.Mission.Values.FirstOrDefault(s => s.MissionID == record.Expedition)?.MapAreaID;
+		DisplayID = KCDatabase.Instance.Mission.Values.FirstOrDefault(s => s.MissionID == record.Expedition)?.DisplayID;
+		Fleet = record.Fleet.MakeFleet();
+		ItemList = response!.ApiUseitemFlag;
```

Comment:
You can remove the null-forgiving operator (!) here because the response should never be null. Update the rest of the code accordingly.